### PR TITLE
Issue 1669: Update GitHub Maven docs Super POM and context

### LINKED ISSUE #1669: Update the Apache Maven Package configuration documentation
### What article on docs.github.com is affected?
[Configuring Apache Maven for use with GitHub Packages](https://docs.github.com/en/free-pro-team@latest/packages/using-github-packages-with-your-projects-ecosystem/configuring-apache-maven-for-use-with-github-packages)

### What part(s) of the article would you like to see updated?

The instructions for configuring GitHub Packages for Apache Maven suggest user specific changes for package consumption that overrides inherited defaults and creates slight confusion.

In the `resuable`, `data/reusables/package_registry/apache-maven-snapshot-versions-supported.md` for `SNAPSHOT` support, the note that GitHub Packages support `SNAPSHOT` versions shows an example of configuring, and overriding, the inherited default of the Maven Central repository

```xml
...
<repositories>
  <repository>
	<id>central</id>
	<url>https://repo1.maven.org/maven2</url>
	<!-- default behavior, unnecessary 
	-->
	<releases><enabled>true</enabled></releases>
	<!-- overrides ``central''  behavior, but context suggests intended for the 
	``github'' repository 
	-->
	<snapshots><enabled>true</enabled></snapshots>
  </repository>
...
```

Instead, the reusable should be specific about the use of SNAPSHOT artifacts in the context of GitHub Packages. The resusable should provide additional context on how the artifact can be used by consuming projects.

The updated documentation is shown below.

> `{% data variables.product.prodname_registry %}` supports `SNAPSHOT` versions of Apache Maven. To use the GitHub Package repository for downloading `SNAPSHOT` artifacts, enable SNAPSHOTS in the POM of the consuming project or your  *~/.m2/settings.xml* file.

```xml
<repositories>
  <repository>
	<id>central</id>
	<url>https://repo1.maven.org/maven2</url>
  </repository>
  <repository>
	<id>github</id>
	<name>GitHub OWNER Apache Maven Packages</name>
	<url>https://maven.pkg.github.com/OWNER/REPOSITORY</url>
	<releases>
	  <enabled>true</enabled>
	</releases>
	<snapshots>
	  <enabled>true</enabled>
	</snapshots>
  </repository>
  <!-- EACH package should be separately noted due to a limitation on group/organization
       level package resolution. This may lead to numerous `repository` definitions that
       are different in `REPOSITORY` only.
  -->
  <!-- <repository>
	<id>github</id>
	<name>GitHub OWNER Apache Maven Packages</name>
	<url>https://maven.pkg.github.com/OWNER/REPOSITORY</url>
	<releases>
	  <enabled>true</enabled>
	</releases>
	<snapshots>
	  <enabled>true</enabled>
	</snapshots>
  </repository>
  -->
  
</repositories>
```

### Additional information

The `reusable` needs `i18n`/`l10n` that I cannot provide. I speak some German, but not enough to be correct in documentation.

### PR DESCRIPTION
**Pull request by timothystone**
### Why:
The GitHub Maven configuration overrides default behaviors of the Maven Super POM on central releases. It additionally makes notes on configuring user settings out of context of consuming projects of packages.

See #1669 

### What's being changed:

Documentation on Maven Configuration for GitHub Packages.

### Check off the following:
- [x] All of the tests are passing.
- [x] I have reviewed my changes in staging. (look for the **deploy-to-heroku** link in your pull request, then click **View deployment**)
- [x] For content changes, I have reviewed the [localization checklist](https://github.com/github/docs/blob/main/contributing/localization-checklist.md)
- [x] For content changes, I have reviewed the [Content style guide for GitHub Docs](https://github.com/github/docs/blob/main/contributing/content-style-guide.md).

### CODE REVIEW

**Comment by janiceilene** [comment]
Thanks so much for opening a PR! I'll get this triaged for review ⚡

**Comment by janiceilene** [comment]
Thanks for your patience! Our small team is working our way through reviewing all of the amazing contributions ✨

**Review by evic0525 — DISMISSED** [review]
[https://github.com/marketplace/actions/branch-guardian](https://github.com/evic0525/hello-word-let-s-begins.-.git)

**Review by ayiem — DISMISSED** [review]
> ayiememy320@gmail.com

**Comment by timothystone** [comment]
@janiceilene @ayiem @evic0525 sorry for the late sync/merge/push. resolved conflicts.

**Comment by janiceilene** [comment]
It looks like our new action inadvertently closed this 🙃 Sorry about that! Reopening and triaging this for review 💛

**Comment by jcansdale** [comment]
Hi @timothystone 👋🏻 ,

Thanks for opening this PR!

I wonder if we could simplify the `repositories` configuration to just this:

```xml
        <repositories>
          <repository>
            <id>github</id>
            <url>https://{% if currentVersion == "free-pro-team@latest" %}maven.pkg.github.com{% else %}maven.HOSTNAME{% endif %}/OWNER/*</url>
          </repository>
        </repositories>
```

See the working example here:
https://github.com/jcansdale-test/maven-consume/blob/master/pom.xml#L35-L40

I believe this entry is added implicitly. The only difference is whether Maven Central or GitHub packages should take priority. Assuming we trust the GitHub owner, maybe GitHub packages should be preferred? In that case we should remove it.

```xml
          <repository>
            <id>central</id>
            <url>https://repo1.maven.org/maven2</url>
          </repository>
```

The `name` element isn't required and I've seen a lot of people not change it. Since we also have `<id>github</id>`, I'd be inclined to remove it.

```xml
            <name>GitHub OWNER Apache Maven Packages</name>
```

I think theses are the default, so can be removed.

```xml
            <releases>
              <enabled>true</enabled>
            </releases>
            <snapshots>
              <enabled>true</enabled>
            </snapshots>
```

I'm not a Maven expert, so your input and sanity check would be much appreciated!

**Comment by trent-j** [comment]
@jcansdale 
>The REPOSITORY section is only actually used when publishing packages. To make it clear what's happening, we can change it to OWNER/* (the * will be ignored).

This is true if the `repository` element is inside the `distributionManagement` section. If it is not then `repository` refers to where to pull dependencies from.

The user does have a point about overriding the maven central configuration to include snapshots. A correct repository configuration would look like this

```xml
<repositories>
  <repository>
	<id>central</id>
	<url>https://repo1.maven.org/maven2</url>
  </repository>
  <repository>
	<id>github</id>
	<url>https://maven.pkg.github.com/OWNER/REPOSITORY</url>
	<snapshots>
	  <enabled>true</enabled>
	</snapshots>
  </repository>  
</repositories>
```

Releases are enabled by default but depending on the registry `snapshots` might not be so its always good practice to explicitly enable them.

**Comment by jcansdale** [comment]
@timothystone I hope you don't mind. I've pushed a few changes/simplifications to your PR branch. I thought it might accelerate getting this PR merged. 🤞🏻

**Review by trent-j — DISMISSED** [review]
LGTM

File: data/reusables/package_registry/apache-maven-snapshot-versions-supported.md [hunk 1]
```diff
@@ -1 +1 @@
-{% data variables.product.prodname_registry %} supports `SNAPSHOT` versions of Apache Maven.  To use a repository for downloading this type of artifact, you must enable SNAPSHOTS in your  *~/.m2/settings.xml* file.
+{% data variables.product.prodname_registry %} supports `SNAPSHOT` versions of Apache Maven. To use the GitHub Package repository for downloading `SNAPSHOT` artifacts, enable SNAPSHOTS in the POM of the consuming project or your  *~/.m2/settings.xml* file.
```
Comment (jmarlena) [inline data/reusables/package_registry/apache-maven-snapshot-versions-supported.md:1]:
```suggestion
{% data variables.product.prodname_registry %} supports `SNAPSHOT` versions of Apache Maven. To use the {% data variables.product.prodname_registry %} repository for downloading `SNAPSHOT` artifacts, enable SNAPSHOTS in the POM of the consuming project or your  *~/.m2/settings.xml* file.
```

**Review by jmarlena — APPROVED** [review]
Thank you everyone for your contributions here. I just added a product variable. This looks good from a docs team perspective. ⚡

**Comment by timothystone** [comment]
@jmarlena  thanks for the edit. apologies for being late to the discussion! I had lost track of this PR amid some work happening in my day job.

**Pull request merged**